### DISCUSSION TITLE
struggle with syntaxError about named exports

### DISCUSSION THREAD

**Discussion by Rvtea** [vitest-dev/vitest#1360] [Q&A]
I struggle with below error all day while do not have any clue yet, so leave a discussion here for some insights if you may have.

```
SyntaxError: Named export 'map' not found. The requested module 'lodash' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'lodash';
const { isNaN, isNumber, isUndefined, map } = pkg;
```

Let me know if else thing I need to provide.

vitest: 0.12.4 / 0.12.9
ts: 4.6.2
yarn: v3

**Comment by Jinjiang** [comment]
Met the same issue. Here is my minimized reproduction
https://github.com/Jinjiang/reproductions/tree/vitest-apollo-20231214

The package in my case is `@apollo/client`. In source code `import { ... } from ...` works. However, when it's written in a file inside `node_modules`, it doesn't work unless using the `import pkg ...` statement.

Is there anyway we can make it works?

Thanks.

**Reply by Jinjiang** [reply]
Update: here is the following issue https://github.com/vitest-dev/vitest/issues/4745 in short, `server.deps.inline` in vitest config helps.

**Reply by JBhrayn** [reply]
May I know how you did it? I followed the instructions but still doesn't work. Would you be able to provide a snippet of the code?
I'm currently facing the same issue, Thanks!:
![image](https://github.com/vitest-dev/vitest/assets/110586557/c946bf34-2fe4-42b4-b784-8cc8cc86bc5e)

**Reply by Jinjiang** [reply]
Sorry for the late reply. Here it is
https://github.com/Jinjiang/reproductions/tree/vitest-apollo-20231214-fix

**Comment by charlieforward9** [comment]
I am having this issue with aws-amplify.
```
SyntaxError: Named export 'Sha256' not found. The requested module '@aws-crypto/sha256-js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@aws-crypto/sha256-js';
const { Sha256 } = pkg;
```

I am not even using this specific module in the tests, so I do not understand why it is preventing me from seeing my tests...

**Comment by charlieforward9** [comment]
It also happens when I simply run `npm run test` with `test: vitest` in the package.json so it looks like this is more than just a VSCode extension issue.

**Comment by JakobMiksch** [comment]
In my case it helped to add this to vite config:

````js
{  
  [...], 
  server: {
    deps: {
      inline: ["vuetify", "vue-pdf-embed"]; // <-- ADD YOUR PROBLEMATIC LIBRARIES HERE
    }
  }
}
````

**Reply by JBhrayn** [reply]
May I know how you did it? I followed the instructions but still doesn't work. Would you be able to provide a snippet of the code?
I'm currently facing the same issue, Thanks!:
![image](https://github.com/vitest-dev/vitest/assets/110586557/9e5140ab-57dc-4801-8b24-e907da1cfdee)
When I add '@tamagui' on the inline: I get another error instead

![image](https://github.com/vitest-dev/vitest/assets/110586557/c81501cf-be1a-4431-a707-e5346d43533f)

**Reply by JakobMiksch** [reply]
I only changed the vite config, that's all.

**Comment by xfire5000** [comment]
{
   ...
   ssr: {
       noExternal: ['v-viewer']
   }
}

**Reply by rgmvisser** [reply]
Thank you sir!

**Comment by bizz84** [comment]
I've just come across this issue on one of my projects (which uses Astro):

```
import {zwitch} from 'zwitch'
        ^^^^^^
SyntaxError: Named export 'zwitch' not found. The requested module 'zwitch' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:
import pkg from 'zwitch';
const {zwitch} = pkg;
```

Based on the comments above, I tried adding this in my `astro.config.mjs`:

```
	vite: {
		plugins: [basicSsl()],
		server: {
			https: true,
			deps: {
				inline: ["zwitch", "hast-util-to-html"],
			}
		},
	},
```

But I still get the same error. Any idea how I could fix it?
